sdk/core/Azure.Core/src/HttpMessage.cs

```diff
@@ -194,6 +198,20 @@ public void SetProperty(string name, object value)
             }
         }
 
+        /// <summary>
+        /// Transfers dispose ownership of the message's Response property to the
+        /// holder of the Response.  After calling this method, the <see cref="Response"/>
+        /// property will still hold the response object, but when this object is
+        /// disposed, it will no longer dispose the response as well.
+        /// </summary>
+        public Response TransferResponseDisposeOwnership()
+        {
+            Response response = Response;
+            _response = null;
```

Comment:
Doesn't setting this to null mean that we effectlivey [no longer hold the response object](https://github.com/Azure/azure-sdk-for-net/blob/81892e9d7e7927eb061bf04ebcac9eeffbec0f6f/sdk/core/Azure.Core/src/HttpMessage.cs#L49-L52)?

Reply:
I should have clarified that the doc comments for this method says that we still hold the response object (which appears to be incorrect).

Reply:
I will update the message for the exception you linked to clarify.

Reply:
Ah, good catch!

Reply:
Updated comment